src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -124,23 +125,34 @@ public override bool Execute()
 
         #region Private methods
 
-        internal /* for testing */ static bool ShouldMergeAnalysisSettings(string language, AnalysisConfig config)
+        internal /* for testing */ static bool ShouldMergeAnalysisSettings(string language, AnalysisConfig config,
```

Comment:
This method isn't pure any longer as it now logs output. However, the output is important.